tests/arrays_test_generated.h

```diff
@@ -0,0 +1,411 @@
+  TestEnum_MIN = TestEnum_A,
+  TestEnum_MAX = TestEnum_C
+};
+
+inline const TestEnum (&EnumValuesTestEnum())[3] {
+  static const TestEnum values[] = {
+    TestEnum_A,
+    TestEnum_B,
+    TestEnum_C
+  };
+  return values;
+}
+
+inline const char * const *EnumNamesTestEnum() {
+  static const char * const names[4] = {
+    "A",
+    "B",
+    "C",
+    nullptr
+  };
+  return names;
+}
+
+inline const char *EnumNameTestEnum(TestEnum e) {
+  if (e < TestEnum_A || e > TestEnum_C) return "";
+  const size_t index = static_cast<size_t>(e);
+  return EnumNamesTestEnum()[index];
+}
+
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) NestedStruct FLATBUFFERS_FINAL_CLASS {
+ private:
+  int32_t a_[2];
+  int8_t b_;
+  int8_t c_[2];
+  int8_t padding0__;
+
+ public:
+  static const flatbuffers::TypeTable *MiniReflectTypeTable() {
+    return NestedStructTypeTable();
+  }
+  NestedStruct() {
+    memset(static_cast<void *>(this), 0, sizeof(NestedStruct));
+  }
+  NestedStruct(const int32_t *_a, TestEnum _b, const TestEnum *_c)
+      : b_(flatbuffers::EndianScalar(static_cast<int8_t>(_b))),
+        padding0__(0) {
+    for (flatbuffers::uoffset_t i = 0; i < 2; i++)
+      reinterpret_cast<flatbuffers::Array<int32_t, 2> *>(a_)->Mutate(i, _a[i]);
+    for (flatbuffers::uoffset_t i = 0; i < 2; i++)
+      reinterpret_cast<flatbuffers::Array<TestEnum, 2, int8_t> *>(c_)->Mutate(i, _c[i]);
+    (void)padding0__;
+  }
+  const flatbuffers::Array<int32_t, 2> *a() const {
+    return reinterpret_cast<const flatbuffers::Array<int32_t, 2> *>(a_);
+  }
+  flatbuffers::Array<int32_t, 2> *mutable_a() {
+    return reinterpret_cast<flatbuffers::Array<int32_t, 2> *>(a_);
+  }
+  TestEnum b() const {
+    return static_cast<TestEnum>(flatbuffers::EndianScalar(b_));
+  }
+  void mutate_b(TestEnum _b) {
+    flatbuffers::WriteScalar(&b_, static_cast<int8_t>(_b));
+  }
+  const flatbuffers::Array<TestEnum, 2, int8_t> *c() const {
+    return reinterpret_cast<const flatbuffers::Array<TestEnum, 2, int8_t> *>(c_);
+  }
+  flatbuffers::Array<TestEnum, 2, int8_t> *mutable_c() {
+    return reinterpret_cast<flatbuffers::Array<TestEnum, 2, int8_t> *>(c_);
+  }
+};
+FLATBUFFERS_STRUCT_END(NestedStruct, 12);
+
+inline bool operator==(const NestedStruct &lhs, const NestedStruct &rhs) {
+  return
+      (lhs.a() == rhs.a()) &&
+      (lhs.b() == rhs.b()) &&
+      (lhs.c() == rhs.c());
+}
+
+inline bool operator!=(const NestedStruct &lhs, const NestedStruct &rhs) {
+    return !(lhs == rhs);
+}
+
+
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) ArrayStruct FLATBUFFERS_FINAL_CLASS {
+ private:
+  float a_;
+  int32_t b_[15];
+  int8_t c_;
+  int8_t padding0__;  int16_t padding1__;
+  NestedStruct d_[2];
+
+ public:
+  static const flatbuffers::TypeTable *MiniReflectTypeTable() {
+    return ArrayStructTypeTable();
+  }
+  ArrayStruct() {
+    memset(static_cast<void *>(this), 0, sizeof(ArrayStruct));
+  }
+  ArrayStruct(float _a, const int32_t *_b, int8_t _c, const NestedStruct *_d)
```

Comment:
This constructor and `NestedStruct(const int32_t *_a, TestEnum _b, const TestEnum *_c)` look extrimly dangerous.
Is it really needed in the generated code?
Can a developer write own builder method?

Reply:
I am aware, unfortunately C++ does not support the C99 syntax for required array lengths, i.e. 
`void func(int arr[static 15])`. Do you think we should use `std::array` here?
Own builder methods are not supported, but AFAIK they don't exist on `struct`s either.

Reply:
Why are structs not exposed as basic C structs in addition to accessor methods? Then you can trivially get the size of any array.